frontend/src/metabase/entities/containers/EntityListLoader.jsx

```diff
@@ -159,20 +169,52 @@ export default class EntityListLoader extends React.Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
+    const { keepPreviousList } = this.props;
+    const { previousList } = this.state;
+    if (!keepPreviousList) {
+      return;
+    }
+
+    if (
+      Array.isArray(prevProps.list) &&
+      !_.isEqual(previousList, prevProps.list)
+    ) {
+      this.setState({ previousList: prevProps.list });
+    }
```

Comment:
Maybe use one `if`, and name the conditional block.

```suggestion
    const shouldUpdatePrevPropsList = keepPreviousList &&
      Array.isArray(prevProps.list) &&
      !_.isEqual(previousList, prevProps.list);
      
    if (shouldUpdatePrevPropsList) {
      this.setState({ previousList: prevProps.list });
    }
```

Reply:
Makes sense, thank you! I'll apply the change myself because committing suggestions often leads to broken prettier checks